blockchain/filedao/staging_buffer.go

```diff
@@ -22,49 +22,32 @@ func newStagingBuffer(size uint64) *stagingBuffer {
 	}
 }
 
-func newStagingBufferFromBytes(buf []byte) (*stagingBuffer, error) {
-	pbInfos := &iotextypes.BlockStores{}
-	if err := proto.Unmarshal(buf, pbInfos); err != nil {
-		return nil, err
-	}
-
-	infos := []*block.Store{}
-	for _, v := range pbInfos.BlockStores {
-		info := &block.Store{}
-		if err := info.FromProto(v); err != nil {
-			return nil, err
-		}
-		infos = append(infos, info)
-	}
-
-	return &stagingBuffer{
-		size:   uint64(len(infos)),
-		buffer: infos,
-	}, nil
-}
-
 func (s *stagingBuffer) Get(pos uint64) (*block.Store, error) {
 	if pos >= s.size {
 		return nil, ErrNotSupported
 	}
 	return s.buffer[pos], nil
 }
 
-func (s *stagingBuffer) Put(pos uint64, blk *block.Store) (bool, error) {
+func (s *stagingBuffer) Put(pos uint64, blkBytes []byte) (bool, error) {
 	if pos >= s.size {
 		return false, ErrNotSupported
 	}
+	blk := &block.Store{}
+	if err := blk.Deserialize(blkBytes); err != nil {
```

Comment:
one test fails because a `[]byte` used in `blk` was reused after the block is committed, causing `blk` to become different later on when we retrieve `blk = Get(pos)` to use
in practice, `blk` to be comitted would be a new struct deserialized from network bytes, so this should not happen, but for safety remove that assumption store a new struct in the buffer